src/OrchardCore/OrchardCore.DisplayManagement/Razor/RazorShapeTemplateViewEngine.cs

```diff
@@ -83,14 +83,14 @@ private async Task<IHtmlContent> RenderRazorViewAsync(string viewName, DisplayCo
                     typeof(IViewEngine).FullName));
             }
 
-            var viewEngine = viewEngines[0] as IRazorViewEngine;
+            var viewEngine = viewEngines[0];
 
             var result = await RenderViewToStringAsync(viewName, displayContext.Value, viewEngine);
 
             return new HtmlString(result);
         }
 
-        public async Task<string> RenderViewToStringAsync(string viewName, object model, IRazorViewEngine viewEngine)
+        public async Task<string> RenderViewToStringAsync(string viewName, object model, IViewEngine viewEngine)
```

Comment:
If it's `RazorShapeTemplateViewEngine` I see `IRazorViewEngine` is appropriate choice otherwise I can expect any other view engine. Correct me if I'm wrong?

Reply:
(bear with me, I'm new to Orchard so could be wrong)
Yes, and no :)
Looks like `RazorShapeTemplateViewEngine` has one Razor specific bit of code: the file name extension. It doesn't use anything specific from `IRazorViewEngine`, could work with any ViewEngine. Everything still works after changing the more generic `IViewEngine` interface.
Looked at OrchardCore's codebase, and didn't see any other view engine. IIRC view engines were introduced when Razor was a new thing, so you could use aspx and razor views. Or your own view engine, but I don't know of alternate view engines that really still exist in 2019. Might be wrong, could be on the roadmap...

In it's current state - due to line 86 - if the 0th viewEngine is not an IRazorViewEngine, `viewEngine` variable will be null. And ultimately `FindView` will fail with a NullReferenceException.
For example if you have MiniProfiler activated. It's [ProfilingViewEngine](https://github.com/MiniProfiler/dotnet/blob/6b3a42a65fed15b8f8b35804668fbc2ea429402b/src/MiniProfiler.AspNetCore.Mvc/Internal/ProfilingViewEngine.cs#L11) is an IViewEngine, doesn't know about `IRazorViewEngine`, yet profiles Razor rendering, if your view engine renders Razor.

This code change is just what seemed like the most straight-forward solution to the NullReferenceException.